### DISCUSSION TITLE
Argon health drain effect doesn't animate correctly on fail

### DISCUSSION THREAD

**Discussion by Gabixel** [ppy/osu#25351] [Q&A] [answered]
I've noticed while playing that the Argon health drain effect seems to behave strangely when you fail. It just disappears. I'm not sure if this is fully intentional.

I made a video with the health component isolated so you can see it in two scenarios (with HUD hidden and visible).

https://github.com/ppy/osu/assets/43073074/11684fb1-ef3e-49e1-8a11-e9811bc24cc3

**Comment by peppy** [comment]
Does this only happen if you have the UI hidden?

**Reply by Gabixel** [reply]
Nope. You can see at the end of the video that I tried with the UI visible. I removed other elements from the skin editor just for a clear view of the health bar.

**Comment by peppy** [comment]
@frenzibyte can you look into this? in the original review i didn't quite follow all the logic surrounding this delegate but it worked so i overlooked it intentionally. I think this is going to require your eyes to understand and refactor to make sense.

**Reply by peppy** [reply]
@frenzibyte second call for help here. If not I'm going to have to rewrite the class to try and make it understandable.

**Comment by frenzibyte** [comment, marked as answer]
Resolved in https://github.com/ppy/osu/pull/25592
